Oracle Network Anchor (#30841)
terraform-providers/terraform-provider-azurerm: website/docs/r/oracle_network_anchor.html.markdown

---
subcategory: "Oracle"
layout: "azurerm"
page_title: "Azure Resource Manager: azurerm_oracle_network_anchor"
description: |-
  Manages a Oracle Network Anchor.
---

# azurerm_oracle_network_anchor

> Manages a Oracle Network Anchor.

Reviewer: Let's change the description to be more concise:
```suggestion
Manages an Oracle Network Anchor.
```
Reference: [reference-documentation-standards.md#descriptions](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/reference-documentation-standards.md#descriptions)

* `name` - (Required) The name which should be used for this Oracle Network Anchor. Changing this forces a new Oracle Network Anchor to be created.

* `resource_anchor_id` - (Required) The ID of the Azure Resource Anchor. Changing this forces a new Oracle Network Anchor to be created.

* `resource_group_name` - (Required) The name of the Resource Group where the Oracle Network Anchor should exist. Changing this forces a new Oracle Network Anchor to be created.

* `subnet_id` - (Required) The ID of the Delegated Azure subnet. Changing this forces a new Oracle Network Anchor to be created.

* `zones` - (Required) A list of availability zones for the Network Anchor. Changing this forces a new Oracle Network Anchor to be created.

---

* `dns_forwarding_rule` - (Optional) One or more `dns_forwarding_rule` blocks as defined below. Changing this forces a new Oracle Network Anchor to be created.

* `dns_listening_endpoint_allowed_cidr` - (Optional) A list of CIDRs that are allowed to send requests to the DNS listening endpoint. Changing this forces a new Oracle Network Anchor to be created.

* `oci_backup_cidr_block` - (Optional) Oracle Cloud Infrastructure backup subnet cidr block.

> * `oci_vcn_dns_label` - (Optional) TODO. Changing this forces a new Oracle Network Anchor to be created.

Reviewer: Can we update the `TODO.` text?
```suggestion
* `oci_vcn_dns_label` - (Optional) The DNS label for the Oracle Cloud Infrastructure VCN. If not specified, defaults to the Network Anchor name. Changing this forces a new Oracle Network Anchor to be created.
```

* `oracle_dns_forwarding_endpoint_enabled` - (Optional) Whether to enable the Oracle DNS forwarding endpoint.

* `oracle_dns_listening_endpoint_enabled` - (Optional) Whether to enable the Oracle DNS listening endpoint.

* `oracle_to_azure_dns_zone_sync_enabled` - (Optional) Whether to enable DNS zone sync from OCI to Azure.

* `tags` - (Optional) A mapping of tags which should be assigned to the Oracle Network Anchor.

---

A `dns_forwarding_rule` block supports the following:

* `domain_names` - (Required) Comma-separated domain names.

> * `forwarding_ip_address` - (Required) Forwarding ip address.

Reviewer: Inconsistent capitalization: 'ip' should be capitalized as 'IP' in the `dns_forwarding_rule` block description.
```suggestion
* `forwarding_ip_address` - (Required) Forwarding IP address.
```
Reference: [reference-documentation-standards](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/reference-documentation-standards.md)

* `resource_group_name` - (Required) The name of the Resource Group where the Oracle Network Anchor should exist. Changing this forces a new Oracle Network Anchor to be created.

* `location` - (Required) The Azure Region where the Oracle Network Anchor should exist. Changing this forces a new Oracle Network Anchor to be created.

* `resource_anchor_id` - (Required) The ID of the Azure Resource Anchor. Changing this forces a new Oracle Network Anchor to be created.

* `subnet_id` - (Required) The ID of the Delegated Azure subnet. Changing this forces a new Oracle Network Anchor to be created.

* `zones` - (Required) A list of availability zones for the Network Anchor. Changing this forces a new Oracle Network Anchor to be created.

---

* `dns_forwarding_rule` - (Optional) One or more `dns_forwarding_rule` blocks as defined below. Changing this forces a new Oracle Network Anchor to be created.

* `dns_listening_endpoint_allowed_cidrs` - (Optional) A list of CIDRs that are allowed to send requests to the DNS listening endpoint. Changing this forces a new Oracle Network Anchor to be created.

> * `oci_backup_cidr_block` - (Optional) Oracle Cloud Infrastructure backup subnet cidr block.

Reviewer 2: ```suggestion
* `oci_backup_cidr_block` - (Optional) Oracle Cloud Infrastructure backup subnet CIDR block.
```

Author: Renamed
